### PR TITLE
community, wg, arch: add wg, sig subprojects and charter for S390x and ARM

### PR DESCRIPTION
**Pull request by dhiller**
**What this PR does / why we need it**:

PR #301 added the wg-TEMPLATE as a starting point for creating working groups.

This PR adds two initial wg-arch working groups:
* wg-arch-s390x
* wg-arch-arm
to define chairs and leads on S390x and ARM.

Also it defines subprojects in sig-buildsystem and sig-ci, so that arch wg members can own code.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: wg-arch-arm/charter.md
Comment (dhiller) [inline wg-arch-arm/charter.md:12]: @zhlhahaha hey, here's the charter for the wg arch arm, please chime in here with the things you want to have added, I'll happily integrate them

File: sigs.yaml
Comment (dhiller) [inline sigs.yaml:200]: @zhlhahaha hey, here's the entry for the wg arch arm, please chime in here with the things you want to have added, I'll happily integrate them

File: sigs.yaml
Comment (dhiller) [inline sigs.yaml:178]: @cfilleke @jschintag @vamsikrishna-siddu hey, here's the entry for the wg arch S390x, please chime in here with the things you want to have added, I'll happily integrate them

File: wg-arch-s390x/charter.md
Comment (dhiller) [inline wg-arch-s390x/charter.md:12]: @cfilleke @jschintag @vamsikrishna-siddu hey, here's the charter for the wg arch S390x, please chime in here with the things you want to have added, I'll happily integrate them

File: sigs.yaml
Comment (cfilleke) [reply sigs.yaml:200]: I'd like to mention @awels and @andymcc here since Alex and Andy have a good grasp of multi-arch challenges in K8s & OCP operator implementation.

File: wg-arch-s390x/charter.md
Comment (cfilleke) [reply wg-arch-s390x/charter.md:12]: Thanks for getting this party started, @dhiller !

File: sigs.yaml
Comment (zhlhahaha) [reply sigs.yaml:200]: Thanks, Daniel!

File: wg-arch-arm/charter.md
Comment (zhlhahaha) [reply wg-arch-arm/charter.md:12]: I'm unsure about what should be included in the "In Scope" section. Perhaps we can follow the same outlines used for x86_64.

Regarding the "Out of Scope" section, I believe we currently only support KVM-based emulation on Arm64 servers. Therefore, binary-based emulation and services on Arm32 are not within the scope.

**Review by iholder101 — COMMENTED** [review]
Thank you @dhiller!
See comments below regarding the templates

**Comment by dhiller** [comment]
/hold

PR https://github.com/kubevirt/community/pull/301 was split out of this one, so this needs to be rebased after that one got merged.

FYI @fabiand

File: wg-TEMPLATE/README.md
Comment (dhiller) [reply wg-TEMPLATE/README.md:1]: @iholder101 @lyarwood I agree - we need some definitions around WGs and SIGs. And yes, this basically is a copy - but still I think that scope, non-scope etc. can still be defined alike it is done for SIGs.

File: wg-TEMPLATE/wg-charter-template.md
Comment (dhiller) [reply wg-TEMPLATE/wg-charter-template.md:17]: As this is just a template, I think it might stay as is.

File: wg-arch-s390x/charter.md
Comment (vamsikrishna-siddu) [reply wg-arch-s390x/charter.md:12]: Hi @dhiller I have put the things I want to add for s390x in my branch here. https://github.com/vamsikrishna-siddu/community/commit/b7c6af2d8f128895e961ec3e40897731d9821424 
can you please review them and add those changes in your PR for s390x.
Thanks

File: sigs.yaml
Comment (dhiller) [reply sigs.yaml:255]: @cfilleke so you are suggesting that they might take chair or lead roles in the WG?

@awels @andymcc WDYT? Are you interested taking this on?

Comment (awels) [reply sigs.yaml:255]: Do we have a definition of what chairs/lead responsibilities are? I want to know that before agreeing.

File: sigs.yaml
Comment (chandramerla) [inline sigs.yaml:205]: @dhiller 
A Question fust for my understanding-
Currently it seems these subproject names aren't used in any OWNERS files. So do we need to update/create OWNERS file along side the code files/dirs while we commit the code and hence we don't refer sub-project names anywhere?

**Review by chandramerla — COMMENTED** [review]
Have one question, which I've added inline. Otherwise looks good.

File: sigs.yaml
Comment (dhiller) [reply sigs.yaml:205]: yes, exactly - my intent is to add the subprojects first, and when you add new OWNERS files where you take ownership of code, then the sigs.yaml will be updated with a reference (hopefully by some automation :smile: )

File: sigs.yaml
Comment (dhiller) [reply sigs.yaml:255]: > SIGs must have at least one and ideally two SIG chairs at any given time. SIG chairs are intended to be organizers and facilitators, responsible for the operation of the SIG and for communication and coordination with the other SIGs, the Steering Committee, and the broader community.

[from the Kubernetes SIGs doc](https://github.com/kubernetes/community/blob/master/governance.md#sigs)

if we map that to WG chairs then I figure WG chairs are responsible of WG operation and for communication and coordination with stakeholder SIGs.

@aburdenthehand @fabiand @lyarwood WDYT?

File: sigs.yaml
Comment (chandramerla) [reply sigs.yaml:205]: when you say "the sigs.yaml will be updated with a reference ", do you mean updating the "owners:" section in the subprojects of sigs.yaml?

Comment (dhiller) [reply sigs.yaml:205]: yes

File: sigs.yaml
Comment (cfilleke) [reply sigs.yaml:259]: /lgtm

**Comment by jean-edouard** [comment]
@dhiller is the hold still relavant?

**Comment by dhiller** [comment]
> @dhiller is the hold still relevant?

Thanks for the reminder :) Yeah, I think a rebase is needed here, other than that it should be good to go. @jean-edouard will ping you again after that's done if you're ok with that?

**Comment by dhiller** [comment]
/unhold

@chandramerla @cfilleke @zhlhahaha I have rebased this PR, PTAL, thanks a lot!

@jean-edouard FYI

File: sigs.yaml
Comment (dhiller) [reply sigs.yaml:282]: @awels we recently updated the membership policy by adding some roles here: https://github.com/kubevirt/community/blob/main/membership_policy.md#special-interest-group-sig-chair

Comment (awels) [reply sigs.yaml:282]: If @andymcc agrees I would be okay with being co-chair with him.

**Comment by dhiller** [comment]
@lyarwood since you had been involved for a bit, would you mind taking another look and see whether this is good to go?

Thank you!

**Comment by dhiller** [comment]
@chandramerla :wave: would you mind taking another look here? We should be good from my POV.

File: sigs.yaml
Comment (aburdenthehand) [inline sigs.yaml:271]: @cfilleke - I can't see you as a member in the KubeVirt org. Can you please open a PR in the project-infra repo and add your name to the [orgs.yaml](https://github.com/kubevirt/project-infra/blob/main/github/ci/prow-deploy/kustom/base/configs/current/orgs/orgs.yaml): github/ci/prow-deploy/kustom/base/configs/current/orgs/orgs.yaml

You meet [the criteria](https://github.com/kubevirt/community/blob/main/membership_policy.md) so this is just the formality. I would be happy to sponsor you.

**Review by lyarwood — APPROVED** [review]
Apologies @dhiller I missed your ping to review this again! Assuming everyone becomes a member of the kubevirt community org then this lgtm.

/lgtm

File: sigs.yaml
Comment (dhiller) [reply sigs.yaml:271]: +1 - so would I!

**Comment by chandramerla** [comment]
> @chandramerla 👋 would you mind taking another look here? We should be good from my POV.

Sorry I didn't check the message in time. Changes wise it looks good and anyway it is merged already.

**Pull request merged**
